homeassistant/components/vesync/humidifier.py

```diff
@@ -0,0 +1,161 @@
+"""Support for VeSync humidifiers."""
+import logging
+
+from homeassistant.components.humidifier import HumidifierEntity
+from homeassistant.components.humidifier.const import SUPPORT_MODES
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .common import VeSyncDevice
+from .const import DEV_TYPE_TO_HA, DOMAIN, VS_DISCOVERY, VS_HUMIDIFIERS
+
+_LOGGER = logging.getLogger(__name__)
+
+HUMIDIFIER_MODE_AUTO = "auto"
+HUMIDIFIER_MODE_MANUAL = "manual"
+HUMIDIFIER_MODE_SLEEP = "sleep"
+
+AVAILABLE_MODES = {
+    "Classic300S": [
```

Comment:
Requires other models to be added here such as `LUH-D301S-WEU` and `Dual200S`

Reply:
I won't be able to test those since I don't have them. Do you still want me to add them?